src/sql_utils.h

```diff
@@ -109,6 +109,30 @@ class SQLite3Guard {
     return sqlite3_exec(handle_.get(), sql, callback, cb_arg, NULL);
   }
 
+  bool beginTransaction() {
```

Comment:
This is fine, but it doesn't allow nesting since you're not using named savepoints allowing for `ROLLBACK TRANSACTION TO SAVEPOINT <savepoint>`.

Reply:
And you might not need that, but just noting it here.

Reply:
Thanks for the info. I did a quick test and SQLite indeed shouts when two `BEGIN TRANSACTION` are called in a row.

I'll add a comment about that